File: frontend/src/app/trip/model/Event.ts
export interface Event {
  creatorEmail: string;
  uuid: string;
  location: LocationDescription;
  eventType: string;
  to: string;
  from: string;
  title: string;
  properties: Property[];
  participants: string[];
}

export interface LocationDescription {
  locationDescription: string;
}

export interface Property {
  key: string;
  value: string;
}

export interface CreateEvent {
  tripId: string;
  title: string;
  locationDescription: string;
  from: string;
  to: string;
  type: string;
  properties: Map<string, string>[];
}

export interface UpdateEvent {
  locationDescription: string;
  from: string;
  to: string;
  title: string;
}

File: backend/src/main/java/com/mytrip/demo/domain/Event.java
package com.mytrip.demo.domain;

import com.mytrip.demo.application.persistance.trip.model.event.type.TripEventTypeName;
import lombok.Builder;
import lombok.Value;

import java.time.LocalDate;
import java.util.*;

@Value
@Builder
public class Event {
    UUID uuid;
    String title;
    LocalDate from;
    LocalDate to;
    LocationDetails location;
    String creatorEmail;
    TripEventTypeName eventType;
    Set<Property> properties;
    List<String> participants;

    @Value
    @Builder
    public static class LocationDetails {
        Double latitude;
        Double longitude;
        String locationDescription;
    }

    @Value
    @Builder
    public static class Property {
        String key;
        String value;
    }
}
